### PR TITLE
drivers: mcr20a: change to T sequence only and small improvements

### PR DESCRIPTION
**Pull request by jfischer-no**
~~Always use the T Sequence for mcr20a_tx, since the API currently does not
support the hardware-assisted ACK reception.~~

Issue a error message, if the _mcr20a_read_reg fails.

Do not execute SPI burst read/write if the buffer is too small.

Unlock mutex if set_pan_id, set_short_addr or set_ieee_addr
fail.

Force abort of the sequence when the higher level changes the channel even
though a T or TX sequence is in progress.

update: reworked  on #4094, the driver checks AR bit and uses appropriate TX sequence now

### CODE REVIEW

**Review by tbursztyka — COMMENTED** [review]
Quickly tried the driver, and it does not seem to tx, though it says the contrary (I see nothing in sniffer).
It receives well however.

**Comment by jfischer-phytec-iot** [comment]
We're doing something different, I'm testing all time because it's crashes sporadic in the spu_mcux_transceive. Have you changed anything in prj_mcr20a.conf? Can please push your setup to a branch on your github?

**Comment by tbursztyka** [comment]
I did not change anything. That said, now you mentioned it, I am testing only through frdm_k64f. I might try the usb dongle as well.

**Comment by jfischer-phytec-iot** [comment]
I am testing the USB drivers for Kinetis, and have noticed that when I use J-Link it comes to abnormal behavior, if I use the mbed dap with openocd then everything is good. 😞

**Comment by jfischer-phytec-iot** [comment]
Will be reworked after #4094 is accepted.

**Comment by jfischer-phytec-iot** [comment]
@tbursztyka rebased, please review

**Review by dleach02 — APPROVED** [review]

File: drivers/ieee802154/ieee802154_mcr20a.c [hunk 1]
```diff
@@ -202,12 +203,13 @@ bool _mcr20a_write_burst(struct mcr20a_spi *spi, bool dreg, u16_t addr,
 {
 	bool retval;
 
-	k_sem_take(&spi->spi_sem, K_FOREVER);
-
 	if ((len + 2) > sizeof(spi->cmd_buf)) {
```
Comment (dleach02) [inline drivers/ieee802154/ieee802154_mcr20a.c]: I'm assuming my PR will be applied over this one to fix the condition test here. If not, I would suggest going ahead and fixing the condition test to be >=.

File: drivers/ieee802154/ieee802154_mcr20a.c [hunk 2]
```diff
@@ -231,12 +233,13 @@ bool _mcr20a_write_burst(struct mcr20a_spi *spi, bool dreg, u16_t addr,
 bool _mcr20a_read_burst(struct mcr20a_spi *spi, bool dreg, u16_t addr,
 			u8_t *data_buf, u8_t len)
 {
-	k_sem_take(&spi->spi_sem, K_FOREVER);
-
 	if ((len + 2) > sizeof(spi->cmd_buf)) {
-		SYS_LOG_ERR("Buffer length too large");
+		SYS_LOG_ERR("cmd buffer too small");
```
Comment (dleach02) [inline drivers/ieee802154/ieee802154_mcr20a.c]: ditto on condition test

File: drivers/ieee802154/ieee802154_mcr20a.c (hunk 1 above)
Comment (jfischer-phytec-iot) [reply drivers/ieee802154/ieee802154_mcr20a.c]: @dleach02 can you please pick the 114d671 and rebase your PR #4575 on it, thx

**Pull request merged**